configuration/postgres/postgres.go

```diff
@@ -388,32 +376,23 @@ func validateInput(keys []string) error {
 	return nil
 }
 
-func (p *ConfigurationStore) subscribeToChannel(ctx context.Context, pgNotifyChanList []string, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
+func (p *ConfigurationStore) subscribeToChannel(ctx context.Context, pgNotifyChannel string, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
 	p.configLock.Lock()
 	defer p.configLock.Unlock()
 	var subscribeID string
-	for _, channel := range pgNotifyChanList {
-		pgNotifyCmd := fmt.Sprintf(listenTemplate, channel)
-		if sub, isActive := p.isSubscriptionActive(req); isActive {
-			if oldStopChan, ok := p.subscribeStopChanMap[sub]; ok {
-				close(oldStopChan)
-				delete(p.subscribeStopChanMap, sub)
-				delete(p.ActiveSubscriptions, channel)
-			}
-		}
-		stop := make(chan struct{})
-		subscribeUID, err := uuid.NewRandom()
-		if err != nil {
-			return "", fmt.Errorf("unable to generate subscription id - %w", err)
-		}
-		subscribeID = subscribeUID.String()
-		p.subscribeStopChanMap[subscribeID] = stop
-		p.ActiveSubscriptions[channel] = &subscription{
-			uuid: subscribeID,
-			keys: req.Keys,
-		}
-		go p.doSubscribe(ctx, req, handler, pgNotifyCmd, channel, subscribeID, stop)
+	pgNotifyCmd := fmt.Sprintf(listenTemplate, pgNotifyChannel)
+	stop := make(chan struct{})
+	subscribeUID, err := uuid.NewRandom()
+	if err != nil {
+		return "", fmt.Errorf("unable to generate subscription id - %w", err)
+	}
+	subscribeID = subscribeUID.String()
+	p.subscribeStopChanMap[subscribeID] = stop
+	p.ActiveSubscriptions[subscribeID] = &subscription{
```

Comment:
Making `ActiveSubscriptions` with subscribeIDs as key instead of channel, because there could be multiple subscription IDs for a common channel (multiple subscribe calls with same channel), but there would be a unique subscriptionID for a given subscribe call with a specific channel